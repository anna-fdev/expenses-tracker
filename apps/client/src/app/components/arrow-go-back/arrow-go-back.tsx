import * as React from 'react';
import { FC } from 'react';
import { IconButton } from '@mui/material';
import { ArrowBack } from '@mui/icons-material';
import { useNavigate } from 'react-router-dom';

export const ArrowGoBack: FC = () => {
  const navigate = useNavigate();

  return (
    <IconButton
      color="secondary"
      aria-label="navigate back to home"
      sx={{ left: 0, top: 36 }}
      onClick={() => navigate(-1)}
    >
      <ArrowBack />
    </IconButton>
  );
};
