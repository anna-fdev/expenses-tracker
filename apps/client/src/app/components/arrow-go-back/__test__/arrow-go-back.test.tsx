import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import { ArrowGoBack } from '../arrow-go-back';

describe('ArrowGoBack testing', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <ArrowGoBack />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
