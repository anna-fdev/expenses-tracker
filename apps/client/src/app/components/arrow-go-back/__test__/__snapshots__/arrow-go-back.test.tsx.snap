// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowGoBack testing should match snapshot 1`] = `
<div>
  <button
    aria-label="navigate back to home"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeMedium css-i7l6r4-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ArrowBackIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
